frontend/public/components/secrets/create-secret/usePullSecretCredentialEntries.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { DockerConfigCredential, DockerConfigData, PullSecretData } from './PullSecretForm';
+import { Base64 } from 'js-base64';
+import * as _ from 'lodash-es';
+import { PullSecretCredential } from './PullSecretCredentialsForm';
+
+const newImageSecretEntry = (): PullSecretCredential => ({
+  address: '',
+  username: '',
+  password: '',
+  email: '',
+  auth: '',
+  uid: _.uniqueId(),
+});
+
+const dockerConfigDataToPullSecretCredentials = (
+  dockerConfigData: DockerConfigData,
+): PullSecretCredential[] => {
+  const entries = Object.entries<DockerConfigCredential>(dockerConfigData ?? {}).map(
+    ([key, value]) => {
+      const decodedAuth = Base64.decode(value?.auth || '');
+      const [parsedUsername, parsedPassword] = decodedAuth?.split(':') ?? [];
+      return {
+        address: key,
+        username: parsedUsername,
+        password: parsedPassword,
+        email: value?.email || '',
+        auth: value?.auth,
+        uid: _.uniqueId(),
+      };
+    },
+  );
+  return entries.length ? entries : [newImageSecretEntry()];
+};
+
+const pullSecretCredentialsToDockerConfigData = (
+  pullSecretCredentials: PullSecretCredential[],
+): DockerConfigData =>
+  pullSecretCredentials.reduce(
+    (acc, { address, ...entry }) => ({
+      ...acc,
+      [address]: entry,
+    }),
+    {},
+  );
+
+export const usePullSecretCredentialEntries = (
+  stringData: string,
+  onChange: (changeData: PullSecretData) => void,
+  onError: (error: any) => void,
+): any => {
+  let dockerConfigData;
+  try {
+    dockerConfigData = JSON.parse(stringData);
+  } catch (err) {
+    dockerConfigData = {};
+    onError(`Error parsing secret's data: ${err.message}`);
+  }
+  const initialEntries = dockerConfigDataToPullSecretCredentials(dockerConfigData);
```

Comment:
This could all be wrapped into on logical memo function. It could arguably be included in the conversion function. Also, don't forget the json in the pull secret could either be .dockerconfigjson or .dockercfg
 
```suggestion
  const initialEntries = React.useMemo(() => {
    try {
      const dockerConfigData = JSON.parse(stringData);
      return dockerConfigDataToPullSecretCredentials(dockerConfigData?.auths || dockerConfigData);
    } catch (err) {
      onError(`Error parsing pull secret: ${err.message}`);
      return [];
    }
  }
```